controllers/deactivation/deactivation_controller.go

```diff
@@ -48,6 +50,7 @@ type Reconciler struct {
 // Note:
 // The Controller will requeue the Request to be processed again if the returned error is non-nil or
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+// nolint: gocyclo
```

Comment:
why?

Reply:
Have you tried to re-enable it now? The code is less complex now then the initial one.

Reply:
Yes, the linter still doesn't like it.

Reply:
have you considered to refactor the code and split the method into multiple ones? TBH, the linter is correct that the method is really complex and should be simplified/refactored.